rapids/rapids.sh

```diff
@@ -1,101 +1,64 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This initialization action script will install rapids on a Dataproc
+# cluster.
+
 set -euxo pipefail
 
 # Detect dataproc image version from its various names
 if (! test -v DATAPROC_IMAGE_VERSION) && test -v DATAPROC_VERSION; then
   DATAPROC_IMAGE_VERSION="${DATAPROC_VERSION}"
 fi
 
-function os_id() {
-  grep '^ID=' /etc/os-release | cut -d= -f2 | xargs
-}
-
-function os_version() {
-  grep '^VERSION_ID=' /etc/os-release | cut -d= -f2 | xargs
-}
-
-function os_codename() {
-  grep '^VERSION_CODENAME=' /etc/os-release | cut -d= -f2 | xargs
-}
-
-function is_rocky() {
-  [[ "$(os_id)" == 'rocky' ]]
-}
-
-function is_ubuntu() {
-  [[ "$(os_id)" == 'ubuntu' ]]
-}
-
-function is_ubuntu20() {
-  is_ubuntu && [[ "$(os_version)" == '20.04'* ]]
-}
-
-function is_ubuntu22() {
-  is_ubuntu && [[ "$(os_version)" == '22.04'* ]]
-}
-
-function is_debian() {
-  [[ "$(os_id)" == 'debian' ]]
-}
-
-function is_debian11() {
-  is_debian && [[ "$(os_version)" == '11'* ]]
-}
-
-function is_debian12() {
-  is_debian && [[ "$(os_version)" == '12'* ]]
-}
-
-function os_vercat() {
-  if is_ubuntu ; then
-      os_version | sed -e 's/[^0-9]//g'
-  elif is_rocky ; then
-      os_version | sed -e 's/[^0-9].*$//g'
-  else
-      os_version
-  fi
-}
-
 function get_metadata_attribute() {
   local -r attribute_name=$1
-  local -r default_value=$2
+  local -r default_value="${2:-}"
   /usr/share/google/get_metadata_value "attributes/${attribute_name}" || echo -n "${default_value}"
 }
 
-readonly DEFAULT_DASK_RAPIDS_VERSION="23.12"
-readonly RAPIDS_VERSION=$(get_metadata_attribute 'rapids-version' ${DEFAULT_DASK_RAPIDS_VERSION})
-
 readonly SPARK_VERSION_ENV=$(spark-submit --version 2>&1 | sed -n 's/.*version[[:blank:]]\+\([0-9]\+\.[0-9]\).*/\1/p' | head -n1)
-readonly DEFAULT_SPARK_RAPIDS_VERSION="22.10.0"
+readonly DEFAULT_SPARK_RAPIDS_VERSION="24.06.1"
 
 if [[ "${SPARK_VERSION_ENV%%.*}" == "3" ]]; then
-  readonly DEFAULT_CUDA_VERSION="11.8"
+  readonly DEFAULT_CUDA_VERSION="12.4"
   readonly DEFAULT_XGBOOST_VERSION="2.0.3"
   readonly SPARK_VERSION="${SPARK_VERSION_ENV}"
+  readonly DEFAULT_XGBOOST_GPU_SUB_VERSION=""
 else
   readonly DEFAULT_CUDA_VERSION="10.1"
   readonly DEFAULT_XGBOOST_VERSION="1.0.0"
   readonly DEFAULT_XGBOOST_GPU_SUB_VERSION="Beta5"
   readonly SPARK_VERSION="2.x"
 fi
 
+# RAPIDS config
+readonly CUDA_VERSION=$(get_metadata_attribute 'cuda-version' ${DEFAULT_CUDA_VERSION})
+function is_cuda12() { [[ "${CUDA_VERSION%%.*}" == "12" ]] ; }
+function is_cuda11() { [[ "${CUDA_VERSION%%.*}" == "11" ]] ; }
+
+readonly DEFAULT_DASK_RAPIDS_VERSION="24.08"
+readonly RAPIDS_VERSION=$(get_metadata_attribute 'rapids-version' ${DEFAULT_DASK_RAPIDS_VERSION})
```

Comment:
Just so I understand, how does this work in terms of CUDA version selection?

Reply:
customer uses a command such as the following to create a cluster.  The customer can select the CUDA version by specifying the 'cuda-version' metadata argument.  We offer a default if they do not select one.  I've been testing with 12.4, but I'm happy to run the test suite with the CUDA version set higher.

Example cluster creation command:
```
gcloud dataproc clusters ${CLUSTER_NAME} \
	--num-masters=1 \
	--num-workers=2 \
	--image-version=2.2-rocky9 \
	--public-ip-address \
	--no-shielded-secure-boot \
	--no-shielded-secure-boot \
	--initialization-actions="gs://${BUCKET}/gpu/install_gpu_driver.sh,gs://${BUCKET}/dask/dask.sh,gs://${BUCKET}/rapids/rapids.sh" \
	--initialization-action-timeout=30m \
	--metadata=gpu-driver-provider=NVIDIA,rapids-runtime=DASK \
	--master-accelerator=type=nvidia-tesla-t4 \
	--worker-accelerator=type=nvidia-tesla-t4 \
	--master-machine-type=n1-standard-16 \
	--worker-machine-type=n1-standard-16 \
	--master-boot-disk-size=100GB \
	--worker-boot-disk-size=100GB \
	--master-boot-disk-type=pd-ssd \
	--worker-boot-disk-type=pd-ssd \
	--format=json \
	--region=us-central1 \
	--max-age=30m \
	--max-idle=5m
```

Reply:
in the test suite, the automation is performed in rapids/test_rapids.py line 72
https://github.com/cjac/initialization-actions/blob/rapids-20240806/rapids/test_rapids.py#L72